### DISCUSSION TITLE
project relationship

### DISCUSSION THREAD

**Discussion by maggu2810** [hathach/tinyusb#1625] [General]
Hello,

I have been looking for a RNDIS host implementation for the RP2040 (Raspberry Pi Pico).
I found two interesting projects:
* Tiny USB
* CherryUSB

This project (Tiny USB) supports the RP2040, supports RNDIS for the device stack, but miss RNDIS for host stack.
The project CherryUSB supports RNDIS for the device stack, supports RNDIS for the host stack, but miss support for the RP2040.

The architecture itself is very similar between the two projects.

Please allow me the question:
Is there a common history of the two projects?
If the one a fork of the other?
Is the other one a takeover of the other, ...?
Do you work together?
Do you "know" each other?

The "Apache-2.0 license" (stated to be used by CherryUSB) would allow to get some inspiration how to implement the RNDIS host stack. Does it?

Best regards,
Markus

**Comment by hathach** [comment]
CherryUSB looks like a great USB stack, more choice is better for user/community. Unfortunately I have no relationship with cherryUSB so far. The folder layout can be similar, but the mechanism is more important. Have you tried both stack to see if the mechanism and API are similar as well ?

**Reply by maggu2810** [reply]
I was looking for a project providing usb host + rndis for rp2040 for a test project.
But currently there has been some little scope change what I need to implement, so I did not have a further look at the code. Sorry
